metric_learn/base_metric.py

```diff
@@ -49,3 +84,45 @@ def transform(self, X=None):
       X = check_array(X, accept_sparse=True)
     L = self.transformer()
     return X.dot(L.T)
+
+
+class SupervisedMixin(object):
+
+  def fit(self, X, y):
+    return NotImplementedError
+
+
+class UnsupervisedMixin(object):
+
+  def fit(self, X, y=None):
+    return NotImplementedError
+
+
+class WeaklySupervisedMixin(object):
```

Comment:
Add `NotImplementedError` for the `__init__` methods of this and the above abstract mixins.

Reply:
Done